canvas_modules/harness/cypress/integration/canvas/palette.js

```diff
@@ -273,3 +273,46 @@ describe("Test nodes in Flyout palette have data-id attribute", function() {
 		cy.verifyNodeHasDataId("Balance", "balance", "Record Ops");
 	});
 });
+
+describe("Test nodes & categories accessibility within palette", function() {
+	beforeEach(() => {
+		cy.visit("/");
+		cy.openCanvasPalette("modelerPalette.json");
+		cy.clickToolbarPaletteOpen();
+	});
+
+	it("Nodes and categories should be accessible", function() {
+		cy.get(".palette-flyout-category")
+			.eq(0)
+			.type("{shift}{alt}Q")
+			// This code simulates the user pressing tab to move the keyboard focus.
+			// TODO - Use the Cypress tab() method when "Native Events" are supported in Cypress
+			.focus();
+
+		cy.verifyCategoryIsOpened("Import");
+
+		cy.get(".bx--accordion__item--active .palette-list-item")
+			.eq(0)
+			.type("{shift}{alt}Q")
+			// This code simulates the user pressing tab to move the keyboard focus.
+			// TODO - Use the Cypress tab() method when "Native Events" are supported in Cypress
+			.focus();
+
+
+		cy.get(".bx--accordion__item--active .palette-list-item")
+			.eq(1)
+			.type("{shift}{alt}Q")
+			// This code simulates the user pressing tab to move the keyboard focus.
+			// TODO - Use the Cypress tab() method when "Native Events" are supported in Cypress
+			.focus();
+
+		cy.get(".bx--accordion__item--active .palette-list-item")
+			.eq(2)
+			.type(" ")
+			// This code simulates the user pressing tab to move the keyboard focus.
+			// TODO - Use the Cypress tab() method when "Native Events" are supported in Cypress
+			.focus();
+
```

Comment:
Please wrapper all the `cy.get()` calls above into lower level calls, like we do with all our other Cypress test cases.

Also, you are hard coding index numbers of accordion items and nodes in your test code. Instead of hard coding index numbers you should use the appropriate label and retrieve the category/node object and then set the focus on that object or simulate the space key press as appropriate. This would make the lower level functions reusable. (And we will probably need them in the future when we build more extensive accessibility tests).

Reply:
Created custom functions for nodes & palette focus action.